test/PowerShellEditorServices.Test/Session/ScriptFileTests.cs

```diff
@@ -580,5 +581,31 @@ public void PropertiesInitializedCorrectlyForUntitled()
                 Assert.Equal(3, scriptFile.FileLines.Count);
             }
         }
+
+        [Fact]
+        public void DocumentUriRetunsCorrectStringForAbsolutePath()
+        {
+            string path;
+            ScriptFile scriptFile;
+            var emptyStringReader = new StringReader("");
+
+            if (Environment.OSVersion.Platform == PlatformID.Win32NT)
+            {
+                path = @"C:\Users\AmosBurton\projects\Rocinate\ProtoMolecule.ps1";
+                scriptFile = new ScriptFile(path, path, emptyStringReader, PowerShellVersion);
+                Assert.Equal("file:///c%3A/Users/AmosBurton/projects/Rocinate/ProtoMolecule.ps1", scriptFile.DocumentUri);
+            }
+            else
+            {
+                // Test the following only on Linux and macOS.
+                path = "/home/AmosBurton/projects/Rocinate/ProtoMolecule.ps1";
+                scriptFile = new ScriptFile(path, path, emptyStringReader, PowerShellVersion);
+                Assert.Equal("file:///home/AmosBurton/projects/Rocinate/ProtoMolecule.ps1", scriptFile.DocumentUri);
+
+                path = "/home/AmosBurton/projects/Rocinate/Proto:Mole:cule.ps1";
```

Comment:
Should we also add a test case for when the path contains at `\` on non-Windows or should that really not matter?

Reply:
Probably but right now VSCode chokes on that test case so not possible to encounter AFAICT from VSCode.  Also the code needs to be updated to support `\` (convert to `%5C`).  Thoughts?

Reply:
System.Uri should already do this?

```
[System.Uri]::new("/user\foo/bar").AbsoluteUri
file:///user%5Cfoo/bar
```

Reply:
Sorry, `Uri` handles this but not `UrlPathEncode`.  Been a bit too focused on that code path.  For the Linux/macOS code path, we use `Uri` so yeah, we should be good. I'll add the test to verify that.